### ISSUE TITLE
renderStyles does not render styles for app.vue dependencies

### ISSUE THREAD

**Issue by peteromano** [nuxt/framework#2553]
### Environment

```
$ npx nuxi info
npx: installed 2 in 2.623s
Nuxt CLI v3.0.0-27338323.1e98259                                                                                                                                                                                                20:33:25
RootDir: /srv/app                                                                                                                                                                                                        20:33:25
Nuxt project info:                                                                                                                                                                                                              20:33:25

------------------------------
- Operating System: `Linux`
- Node Version:     `v14.17.6`
- Nuxt Version:     `latest`
- Package Manager:  `unknown`
- Bundler:          `Webpack`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

👉 Report an issue: https://github.com/nuxt/nuxt.js/issues/new                                                                                                                                                                  20:33:25  

👉 Suggest an improvement: https://github.com/nuxt/nuxt.js/discussions/new

👉 Read documentation: https://nuxtjs.org
```

^ Bundler reports Webpack, but it's actually Vite fyi

Running in Docker node-alpine container

### Reproduction

With some basic test components with scoped styles, run `npx nuxi build`. Nuxt build rolls up the styles into bootstrap CSS file, but nothing seems to load the styles in the browser.

```
export default defineNuxtConfig({
  srcDir: 'context/',
  server: {
    port: process.env.NITRO_PORT
  },
  vite: {
    logLevel: process.env.ENV === 'development' ? 'warn' : 'info',
    server: {
      watch: {
        usePolling: true
      }
    }
  },
  // buildModules: [
  //   '@nuxtjs/pwa'
  // ],
  pwa: {
    manifest: {
      name,
      short_name: name,
      description
    },
    meta: {
      theme_color: theme || undefined,
      name,
      title: name,
      description,
      host
    }
  }
})
```

### Describe the bug

The `bootstrap.{hash}.css` file that the browser runtime loads from `/_nuxt/assets/` reports as having no data, and the rolled up styles never get loaded on the page.

However, A) other assets load fine, and B) hitting the css file URL directly shows the correct output. Also, **dev build is fine**.

Does something need to be configured to tell nuxt to load the styles?

![image](https://user-images.githubusercontent.com/392387/147505606-77daba9b-fbaf-468f-9237-505e07c15934.png)

The following from `.output/server/chunks/app/render.mjs#getSPARenderer` seems to not be getting run:

`renderStyles: () => entryFiles.flatMap(({ css }) => css).filter((css) => css != null).map((file) => `<link rel="stylesheet" href="${publicPath}${file}">`).join(""),`

With manifest data:

```
const client_manifest = {
  "../node_modules/nuxt3/dist/app/entry.mjs": {
    "file": "entry-702f5983.mjs",
    "src": "../node_modules/nuxt3/dist/app/entry.mjs",    
    "isEntry": true,
    "dynamicImports": [
      "../node_modules/nuxt3/dist/app/bootstrap.mjs"      
    ]
  },
  "../node_modules/nuxt3/dist/app/bootstrap.mjs": {       
    "file": "bootstrap-a06256d2.mjs",
    "src": "../node_modules/nuxt3/dist/app/bootstrap.mjs",
    "isDynamicEntry": true,
    "css": [
      "assets/bootstrap.7234385c.css"
    ]
  }
};
```

`curl localhost:8080 | grep stylesheet` on the SSR side also doesn't have any hits.

The output of the log here is empty array:
```
function renderStyles(ssrContext, rendererContext) {
  const { styles } = getRequestDependencies(ssrContext, rendererContext);
        console.log(Object.values(styles))
  return Object.values(styles).map(({ path }) => `<link rel="stylesheet" href="${rendererContext.publicPath}${path}">`).join("");
}
```

and log of getRequestDependencies is: 
```
{
  scripts: {
    '../node_modules/nuxt3/dist/app/entry.mjs': { path: 'entry-702f5983.mjs', type: 'module' }
  },
  styles: {},       <----------------------- WTH????
  preload: {
    '../node_modules/nuxt3/dist/app/entry.mjs': { path: 'entry-702f5983.mjs', type: 'module' }
  },
  prefetch: {
    '../node_modules/nuxt3/dist/app/bootstrap.mjs': { path: 'bootstrap-a06256d2.mjs', type: 'module' },
    'assets/bootstrap.7234385c.css': { path: 'assets/bootstrap.7234385c.css' }
  }
}
```

FYI, there is **NO** `../node_modules` in my container (I wipe them after a build; and there is no `nuxt3` dir in the .output node_modules.)

Going to try without wiping modules and see if that works.

UPDATE: That (^) did not work.

This is looking more like a legit bug: styles are just not rendering.

UPDATE: As a **super** basic nuxt setup, I was using `app.vue`. If using `pages/index.vue`, styles load correctly.

So... **This is either a legit bug that should be fixed, _OR_ app.vue should be documented as totally deprecated (I was under the impression app.vue vs. pages framework was optional)?**


### Additional context

https://github.com/nuxt/framework/blame/main/packages/nitro/src/runtime/app/render.ts
https://github.com/nuxt/framework/commits/main/packages/nitro/src/runtime/app/render.ts

### Logs

_No response_

**Comment by peteromano** [comment]
@pi0 Any help/comment on this? Please?

**Comment by sasaorasanin** [comment]
I have the same error within app.vue.

**Comment by peteromano** [comment]
@sasaorasanin yea what the heck!

@pi0 Any reason why Nuxt 3 just doesn't deprecate app.vue?

**Comment by StevenJPx2** [comment]
I have been facing A similar issue, where CSS styles don't render at build but are all fine in dev. The `nuxt-welcome` component loads the style with no problems in both dev and build though.

**Comment by kms-rscholz** [comment]
Can confirm, the stylesheet is simply not applied. 

The stylesheet is linked in the `head`: `<link rel="prefetch" href="/_nuxt/assets/bootstrap.8de70e74.css">` and also downloaded according to the Network tab. The styles just simply do not get applied.

When changing `rel="prefetch"` to `rel="stylesheet"` manually the styles get applied.

**Comment by kms-rscholz** [comment]
@peteromano 

Adding  the following line to `getAllDependencies()`  fixes the issue for me: 
`Object.assign(allDeps.styles, dynamicDeps.styles);` 

Can you confirm?

```
function getAllDependencies(ids, rendererContext) {
  var _a;
  const cacheKey = Array.from(ids).join(",");
  if (rendererContext._dependencySets[cacheKey]) {
    return rendererContext._dependencySets[cacheKey];
  }
  const allDeps = {
    scripts: {},
    styles: {},
    preload: {},
    prefetch: {}
  };
  for (const id of ids) {
    const deps = getModuleDependencies(id, rendererContext);
    Object.assign(allDeps.scripts, deps.scripts);
    Object.assign(allDeps.styles, deps.styles);
    Object.assign(allDeps.preload, deps.preload);
    Object.assign(allDeps.prefetch, deps.prefetch);
    for (const dynamicDepId of ((_a = rendererContext.clientManifest[id]) == null ? void 0 : _a.dynamicImports) || []) {
      const dynamicDeps = getModuleDependencies(dynamicDepId, rendererContext);
      Object.assign(allDeps.prefetch, dynamicDeps.scripts);
      Object.assign(allDeps.prefetch, dynamicDeps.styles);
      Object.assign(allDeps.styles, dynamicDeps.styles);
      Object.assign(allDeps.prefetch, dynamicDeps.preload);
      Object.assign(allDeps.prefetch, dynamicDeps.prefetch);
    }
  }
  for (const id in allDeps.prefetch) {
    if (id in allDeps.preload) {
      delete allDeps.prefetch[id];
    }
  }
  rendererContext._dependencySets[cacheKey] = allDeps;
  return allDeps;
}
```

**Comment by StevenJPx2** [comment]
For those who need to use Nuxt 3 without this bug, I've noticed that this builds 'v3.0.0-27272804.ca557e7' and 'v3.0.0-27319101.3e82f0f' work.

**Comment by StevenJPx2** [comment]
Any update on this? This seems to be a big issue to not have CSS on build with Vite.

**Comment by milesingrams** [comment]
Same issue, styles do not appear on deployed site.

**Comment by StevenJPx2** [comment]
@danielroe @milesingrams @peteromano can you check if this is fixed with the latest version? "v3.0.0-27383920.81ee59c"

**Comment by tamanobi** [comment]
I have the same error on `v3.0.0-27385496.7a1cf62`.

I did `nuxi build` and `nuxi preview`. Generated files are not styled.

The manifest file I got is below(`.output/server/chunks/app/client.manifest.mjs`).

```javascript
const client_manifest = {
  "node_modules/nuxt3/dist/app/entry.mjs": {
    "file": "entry-b7405ca1.mjs",
    "src": "node_modules/nuxt3/dist/app/entry.mjs",
    "isEntry": true,
    "dynamicImports": [
      "_bootstrap-c038b39b.mjs"
    ]
  },
  "_bootstrap-c038b39b.mjs": {
    "file": "bootstrap-c038b39b.mjs",
    "isDynamicEntry": true,
    "dynamicImports": [
      "pages/index.vue",
      "pages/question.vue"
    ],
    "css": [
      "bootstrap.0954eba3.css"
    ]
  }
};

export { client_manifest as default };
```

I got the following workaround to add a css dependencies on `.output/server/chunks/app/client.manifest.mjs`.

```diff
     "isEntry": true,
     "dynamicImports": [
       "_bootstrap-c038b39b.mjs"
+    ],
+    "css": [
+      "bootstrap.0954eba3.css"
     ]
   },
   "_bootstrap-c038b39b.mjs": {
```

But, I haven't found codes that generate yet.
